src/main/java/com/webcrawling/repositories/webCrawler/WebPageRepository.java

```diff
@@ -1,7 +1,13 @@
 package com.webcrawling.repositories.webCrawler;
 
 import com.webcrawling.entities.webCrawler.WebPage;
+import org.springframework.data.elasticsearch.annotations.Query;
 import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;
+import org.springframework.data.repository.query.Param;
 
-public interface WebPageRepository extends ElasticsearchRepository<WebPage, Long> {
+import java.util.List;
+
+public interface WebPageRepository extends ElasticsearchRepository<WebPage, String> {
+    @Query("{\"bool\": {\"should\": [{\"match\": {\"title\": {\"query\": \"?0\", \"operator\": \"and\"}}}, {\"match\": {\"content\": {\"query\": \"?0\", \"operator\": \"and\"}}}]}}")
+    List<WebPage> searchByTitleOrContent(String filter);
```

Comment:
Its difficult to read and develop query .Please find a way to construct elastic search query manually
https://medium.com/shoutloudz/elastic-search-with-spring-boot-357bb0f57972